rasterio/env.py

```diff
@@ -299,14 +340,14 @@ def delenv():
 def ensure_env(f):
     """A decorator that ensures an env exists before a function
     calls any GDAL C functions."""
-    if local._env:
-        return f
-    else:
-        @wraps(f)
-        def wrapper(*args, **kwds):
-            with Env():
+    @wraps(f)
+    def wrapper(*args, **kwds):
+        if local._env:
+            return f(*args, **kwds)
+        else:
```

Comment:
Before, this test for existence of a GDAL environment only happened the first time `rasterio/env.py` was loaded! This meant that the default config options

```
CHECK_WITH_INVERT_PROJ=True
GTIFF_IMPLICIT_JPEG_OVR=False
```

 would almost always be on (unless explicitly overridden) if we imported the warp, features, fill, or io modules.